From 51b57495d0240a35e7b2ddd3432aa6a6a94cddfc Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:06 -0400
Subject: [ethernet] qlcnic: Fix initialization of vlan list

Message-id: <1408626422-3941-83-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88668
O-Subject: [PATCH RHEL 7.1 082/138] qlcnic: Fix initialization of vlan list.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Rajesh Borundia <rajesh.borundia@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id bcf6cb1aa415055749d855eead774896141eb5d8

o Do not re-initialize vlan list in case of adapter reset.

Signed-off-by: Rajesh Borundia <rajesh.borundia@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
index 17a1ca2..0638c18 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
@@ -448,8 +448,7 @@ static int qlcnic_sriov_set_guest_vlan_mode(struct qlcnic_adapter *adapter,
  return 0;
 }
 
-static int qlcnic_sriov_get_vf_acl(struct qlcnic_adapter *adapter,
-       struct qlcnic_info *info)
+static int qlcnic_sriov_get_vf_acl(struct qlcnic_adapter *adapter)
 {
  struct qlcnic_sriov *sriov = adapter->ahw->sriov;
  struct qlcnic_cmd_args cmd;
@@ -495,10 +494,6 @@ static int qlcnic_sriov_vf_init_driver(struct qlcnic_adapter *adapter)
  if (err)
   return -EIO;
 
- err = qlcnic_sriov_get_vf_acl(adapter, &nic_info);
- if (err)
-  return err;
-
  if (qlcnic_83xx_get_port_info(adapter))
   return -EIO;
 
@@ -555,6 +550,10 @@ static int qlcnic_sriov_setup_vf(struct qlcnic_adapter *adapter,
  if (err)
   goto err_out_send_channel_term;
 
+ err = qlcnic_sriov_get_vf_acl(adapter);
+ if (err)
+  goto err_out_send_channel_term;
+
  err = qlcnic_setup_netdev(adapter, adapter->netdev, pci_using_dac);
  if (err)
   goto err_out_send_channel_term;
-- 
1.7.1